File: Back/database/controllers/roleController.js
let Role = require("../models/roles")
let responeData = require("./responeData") 

const create = async (req, res) => {
    try {
        const { name } = req.body;
        var newRole = {
            name: name,
        };

        const role = new Role(newRole);
        role.save()
        .then( async () => {
            console.log("Add user")
            let data = await responeData.createData(0,"","Success")
            return res.status(200).json(data)
        })
        .catch(async (err) => {
            console.log("Err in add user")
            let data =  await responeData.createData(1,"",err)
            return res.status(200).json(data)
        })
    } catch (error) {
        console.log(error)
        return res.status(200).json("error")
    }

};

const getAll = async (req, res) =>{

    try {
        Role.find()
        .then( async (roles)=>{
            // console.log(roles)
            let data =  await responeData.createData(0,roles, process.env.SUCCESS)
            return res.status(200).json(data)
        })
        .catch(async (err)=>{
            // console.log(err)
            let data =  await responeData.createData(1,null,err)
            return res.status(200).json(data)
        })
        
    } catch (error) {
        return res.status(400)
    }

}


module.exports = {create,getAll}

File: netBackEnd/netBackEnd/Area/Admin/Controller/RoleController.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using netBackEnd.Common;
using netBackEnd.Models;
using netBackEnd.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace netBackEnd.Area.Admin
{
    [Area("Admin")]
    [Route("admin/role")]
    [ApiController]
    public class RoleController : ControllerBase
    {
        private RoleService role = new RoleService();

        [HttpPost("create")]
        public IActionResult createUser([FromBody] Roles newRole)
        {

            var result = role.createRole(newRole);

            if (result)
            {
                var data = CreateData.create(0, null, "Success");
                return Ok(data);
            }
            else
            {
                var data = CreateData.create(1, null, "Error");
                return Ok(data);
            }


        }

        [HttpGet("get")]
        public IActionResult getAll()
        {

            //var rng = new Random();
            //return Enumerable.Range(1, 5).Select(index => new WeatherForecast
            //{
            //    Date = DateTime.Now.AddDays(index),
            //    TemperatureC = rng.Next(-20, 55),
            //    Summary = Summaries[rng.Next(Summaries.Length)]
            //})
            //.ToArray();
            var list = role.getAllAsync();
            var data = CreateData.create(0, list.Result, "Success");
            return Ok(data);
        }

    }
}
